runtime/src/migration.rs

```diff
@@ -25,110 +25,223 @@ use sp_runtime::TryRuntimeError;
+					log::info!("No agenda at block: {:?}", block);
+				}
+			}
+
+			let migration_weight =
+				pallet_scheduler::migration::v3::MigrateToV4::<Runtime>::on_runtime_upgrade();
+
+			weight.saturating_add(migration_weight)
+		}
+
+		#[cfg(feature = "try-runtime")]
+		pub fn post_upgrade(state: Vec<u8>) -> Result<(), TryRuntimeError> {
+			let (agendas, corrupted_agendas_blocks): (u32, u32) =
+				Decode::decode(&mut &state[..]).expect("Encoded data is always a valid tuple; qed");
+			let valid_agendas = agendas - corrupted_agendas_blocks;
+			let current_agendas = pallet_scheduler::Agenda::<Runtime>::iter().count() as u32;
+
+			ensure!(
+				current_agendas == valid_agendas,
+				"Corrupted agendas did not get cleared"
+			);
+			let encoded_data = (valid_agendas).encode();
+			pallet_scheduler::migration::v3::MigrateToV4::<Runtime>::post_upgrade(encoded_data)?;
+
+			Ok(())
+		}
 	}
+}
 
-	#[cfg(feature = "try-runtime")]
-	pub(crate) fn pre_upgrade() -> Result<Vec<u8>, &'static str> { Ok(sp_std::vec![]) }
+mod bounties {
+	use super::*;
 
-	#[cfg(feature = "try-runtime")]
-	pub(crate) fn post_upgrade(_state: Vec<u8>) -> Result<(), &'static str> {
-		use crate::constants::nomination_pools::*;
-
-		ensure!(
-			MinJoinBond::<Runtime>::get() == MIN_JOIN_BOND,
-			"Expected `nomination_pools::MinJoinBond == 1 * AVL`"
-		);
-		ensure!(
-			MinCreateBond::<Runtime>::get() == MIN_CREATE_BOND,
-			"Expected `nomination_pools::MinCreateBond == 10 * AVL`"
-		);
-		ensure!(
-			MaxPools::<Runtime>::get() == Some(MAX_POOLS),
-			"Expected `nomination_pools::MaxPools == 16`"
-		);
-		ensure!(
-			MaxPoolMembersPerPool::<Runtime>::get() == Some(MAX_MEMBERS_PER_POOL),
-			"Expected `nomination_pools::MaxPoolMembersPerPool == 100`"
-		);
-		ensure!(
-			MaxPoolMembers::<Runtime>::get() == Some(MAX_MEMBERS),
-			"Expected `nomination_pools::MaxMembers == 1600`"
-		);
+	// Migrations that set `StorageVersion`s which was missed to set.
+	pub mod v1_to_v4 {
+		use super::*;
 
-		Ok(())
+		#[cfg(feature = "try-runtime")]
+		pub fn pre_upgrade() -> Result<Vec<u8>, &'static str> {
+			let storage_version = Bounties::on_chain_storage_version();
+			log::info!("Old bounties storage version: {:?}", storage_version);
+			Ok(sp_std::vec![])
+		}
+
+		// Here we update the storage version to be consistent even if pallet was initialized correctly
+		pub fn on_runtime_upgrade() -> Weight {
+			let storage_version = Bounties::on_chain_storage_version();
+			if storage_version < 4 {
+				StorageVersion::new(4).put::<Bounties>();
+			}
+
+			<Runtime as frame_system::Config>::DbWeight::get().reads_writes(1, 1)
+		}
+
+		#[cfg(feature = "try-runtime")]
+		pub fn post_upgrade(_: Vec<u8>) -> Result<(), &'static str> {
+			let storage_version = Bounties::on_chain_storage_version();
+			log::info!("New bounties storage version: {:?}", storage_version);
+			Ok(())
+		}
 	}
+}
 
-	mod v0_to_v1 {
+mod nomination_pools {
+	use super::*;
+
+	#[allow(dead_code)]
+	pub mod v0_to_v1 {
```

Comment:
Do we need this? I think once both our networks are in sync, we can just clear the upstream pallet migrations from our code.

Reply:
It's good practice to keep it as parity is doing it i guess, but it's not used indeed

Reply:
What I'm trying to convey is that for pallets that are part of the upstream, we can remove them once there are no more migrations required on any chain. However, for our own pallets, we can choose to keep them.

Reply:
Agreed, i was just wondering if there was any reason to keep it ? (did not really touch it since it's not my code)
Any thought @MiguelDD1